pkg/instancetype/webhooks/vm/admitter_test.go

```diff
@@ -0,0 +1,85 @@
+package vm_test
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/golang/mock/gomock"
+	k8sv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	virtv1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/client-go/kubevirt/fake"
+
+	webhooks "kubevirt.io/kubevirt/pkg/instancetype/webhooks/vm"
+	"kubevirt.io/kubevirt/pkg/libvmi"
+)
+
+var _ = Describe("Instance type and Preference VirtualMachine Admitter", func() {
+	var virtClient *kubecli.MockKubevirtClient
+	BeforeEach(func() {
+		ctrl := gomock.NewController(GinkgoT())
+		virtClient = kubecli.NewMockKubevirtClient(ctrl)
+
+		virtClient.EXPECT().VirtualMachine(metav1.NamespaceDefault).Return(
+			fake.NewSimpleClientset().KubevirtV1().VirtualMachines(metav1.NamespaceDefault)).AnyTimes()
+
+		virtClient.EXPECT().VirtualMachineInstance(metav1.NamespaceDefault).Return(
+			fake.NewSimpleClientset().KubevirtV1().VirtualMachineInstances(metav1.NamespaceDefault)).AnyTimes()
+
+		virtClient.EXPECT().VirtualMachineInstancetype(k8sv1.NamespaceDefault).Return(
+			fake.NewSimpleClientset().InstancetypeV1beta1().VirtualMachineInstancetypes(metav1.NamespaceDefault)).AnyTimes()
+
+		virtClient.EXPECT().VirtualMachineClusterInstancetype().Return(
+			fake.NewSimpleClientset().InstancetypeV1beta1().VirtualMachineClusterInstancetypes()).AnyTimes()
+
+		virtClient.EXPECT().VirtualMachinePreference(k8sv1.NamespaceDefault).Return(
+			fake.NewSimpleClientset().InstancetypeV1beta1().VirtualMachinePreferences(metav1.NamespaceDefault)).AnyTimes()
+
+		virtClient.EXPECT().VirtualMachineClusterPreference().Return(
+			fake.NewSimpleClientset().InstancetypeV1beta1().VirtualMachineClusterPreferences()).AnyTimes()
+	})
+	Context("Given a VirtualMachine", func() {
+		DescribeTable("should admit when", func(vm *virtv1.VirtualMachine) {
+			admitter := webhooks.NewAdmitter(virtClient)
```

Comment:
Would it be possible to initialize this variable only once and then reuse it for other tests? Or does it store some state?

Reply:
Once more tests are added here sure but for now I'll keep it inline.